contracts/modules/Experimental/TransferManager/KYCTransferManager.sol

```diff
@@ -0,0 +1,102 @@
+pragma solidity ^0.5.0;
+
+import "../../TransferManager/TransferManager.sol";
+import "../../../interfaces/IDataStore.sol";
+import "../../../interfaces/ISecurityToken.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title Transfer Manager module for core transfer validation functionality
+ */
+contract KYCTransferManager is TransferManager {
+
+    using SafeMath for uint256;
+    
+    bytes32 public constant KYC_NUMBER = "KYC_NUMBER"; //We will standardize what key to use for what.
+
+    bytes32 public constant KYC_PROVIDER = "KYC_PROVIDER";
+
+    bytes32 public constant KYC_ARRAY = "KYC_ARRAY";
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress)
+        public
+        Module(_securityToken, _polyAddress)
+    {
+    }
+
+    /**
+     * @notice This function returns the signature of configure function
+     */
+    function getInitFunction() public pure returns (bytes4) {
+        return bytes4(0);
+    }
+
+    function verifyTransfer(address /*_from*/, address _to, uint256 /*_amount*/, bytes memory /* _data */, bool /* _isTransfer */) 
+        public 
+        returns (Result) 
+    {
+        if (!paused) {
+            bytes32 key = _getKYCKey(_to);
+            IDataStore dataStore = IDataStore(ISecurityToken(securityToken).dataStore());
+            if (dataStore.getUint256(key) > 0)
+                return Result.VALID;
+        }
+        return Result.NA;
+    }
+
+    function modifyKYC( address _investor, bool _kycStatus) public withPerm(KYC_PROVIDER) {
+        _modifyKYC(_investor, _kycStatus);
+    }
+
+    function _modifyKYC(address _investor, bool _kycStatus) internal {
+        IDataStore dataStore = IDataStore(ISecurityToken(securityToken).dataStore());
```

Comment:
Yeah, I didn't do that in this PR because I am split if we should make a getter function or just store the data store address in the constructor.
Using the getter will mean higher gas but every time the latest data store will be fetched. Storing it as a variable will mean lower gas but we'll need to update ds address in each module if we change ds.

data store will rarely (if ever) be changed after deployment but storing the data store in every module may end up being more expensive if/when rent is introduced.